src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -744,12 +815,59 @@ private boolean handleMaterials(@NotNull final Block block, @NotNull final IBloc
             return true;
         }
 
-        if (isBlockFree(block, block.getMetaFromState(blockState)))
+        List<ItemStack> itemList = new ArrayList<>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'itemList' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)